src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -9,18 +9,146 @@
+        this.size = size;
+        this.threshold = (int) (keyArray.length * 0.75);
+    }
+
     @Override
     public void put(K key, V value) {
-        
+        int arrayIndex = findIndex(key);
+        if (threshold == size) {
+            reSize();
+        }
+        addNewNode(key, value, arrayIndex);
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        int newArrayIndex = findIndex(key);
+        if (Objects.equals(keyArray[newArrayIndex].key, key)) {
+            return keyArray[newArrayIndex].value;
+        }
+        if (!Objects.equals(keyArray[newArrayIndex].next, null)) {
+            Node<K, V> nextKey = keyArray[newArrayIndex].next;
+            while (nextKey != null) {
+                if (Objects.equals(nextKey.key, key)) {
+                    return nextKey.value;
+                }
+                nextKey = nextKey.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean isEmpty(int arrayIndex) {
+        return Objects.equals(keyArray[arrayIndex], null);
+    }
+
+    private void reSize() {
+        int newArraySize = keyArray.length * 2;
+        Node<K, V>[] newKeyArray = (Node<K, V>[]) new Node[newArraySize];
+        threshold = (int) (newArraySize * 0.75);
+        size = 0;
+        Node<K, V>[] oldKeyArray = keyArray;
+        keyArray = newKeyArray;
+        for (int i = 0; i < oldKeyArray.length; i++) {
+            if (Objects.nonNull(oldKeyArray[i])) {
+                if (Objects.isNull(oldKeyArray[i].next)) {
+                    int newArrayIndex = findIndex(oldKeyArray[i].key);
+                    addNewNode(oldKeyArray[i].key, oldKeyArray[i].value, newArrayIndex);
+                } else {
+                    int newArrayIndex = findIndex(oldKeyArray[i].key);
+                    addNewNode(oldKeyArray[i].key, oldKeyArray[i].value, newArrayIndex);
+                    Node<K, V> nextNode = oldKeyArray[i].next;
+                    while (nextNode != null) {
+                        int newNodeIndex = findIndex(nextNode.key);
+                        addNewNode(nextNode.key, nextNode.value, newNodeIndex);
+                        nextNode = nextNode.next;
+                    }
+                }
+            }
+        }
+    }
+
+    private int findIndex(K key) {
+        if (Objects.isNull(key)) {
+            return 0;
+        }
+        return Math.abs(key.hashCode() % keyArray.length);
+    }
+
+    private void addNewNode(K key, V value, int arrayIndex) {
+
+        if (isEmpty(arrayIndex)) {
+            if (Objects.isNull(key)) {
+                keyArray[arrayIndex] = new Node<>(null, value, 0, null);
+                size++;
+                return;
+            }
+            keyArray[arrayIndex] = new Node<>(key, value, key.hashCode(), null);
+            size++;
```

Comment:
create method hash(key) to calculate hashCode of keys and consider null keys.
And after that you can remove code duplication.

Reply:
@al-pavlenko create & use method hash(key).